projects/plugins/jetpack/require-lib.php

```diff
@@ -20,16 +16,29 @@ function jetpack_require_lib( $slug ) {
 	 * @param string $lib_dir Path to the library directory.
 	 */
 	$lib_dir = apply_filters( 'jetpack_require_lib_dir', $lib_dir );
-	$choices = array(
-		"$lib_dir/$slug.php",
-		"$lib_dir/$slug/0-load.php",
-		"$lib_dir/$slug/$basename.php",
-	);
-	foreach( $choices as $file_name ) {
-		if ( is_readable( $file_name ) ) {
-			require_once $file_name;
-			return;
-		}
+
+	$loaded_key = "{$lib_dir}{$slug}";
+	if ( ! empty( $loaded[ $loaded_key ] ) ) {
+		return;
+	}
+
+	$loaded[ $loaded_key ] = true;
+
+	$file_name = "$lib_dir/$slug.php";
+	if ( is_readable( $file_name ) ) {
+		require_once $file_name;
```

Comment:
Would there be an advantage of using `include_once` instead of `is_readable` and `require_once`? IIRC, `is_readable` has some caching associated with it, but with the memoization, we'd only run over this once.

```
	if ( include_once $file_name ) {
		return;
	}
```

Reply:
Wouldn't that generate spurious warnings about the attempted filename not being found?

Reply:
🤦 You're right. I was just thinking about the performance. IF we can audit/update all of our usages to the same format...

Disregard this from me.